### ISSUE TITLE
Day to Day Change

### ISSUE THREAD

**Issue by trbutler** [antoinevulcain/Financial-Modeling-Prep-API#8]
Is there a simple way to determine how much a stock price has changed (just day-over-day) without downloading the full historical report that may go back decades?

**Comment by antoinevulcain** [comment]
If I understand you just want the stock price change from the last day? You could use this endpoint: `[https://financialmodelingprep.com/api/v3/historical-price-full/AAPL?timeseries=2](https://financialmodelingprep.com/api/v3/historical-price-full/AAPL?timeseries=2)` it would show you only the change for the past day

**Comment by trbutler** [comment]
Thank you! Is there a way to do that while also getting the current price in the same query? E.g. so it can be parsed to report the amount change of price over the previous close. (For example, with AAPL, it is up $2.47 today over its previous close; the "profile" API's "change" information seems to be referring to something else.) I could do that by using the query you gave and the "realtime" query and then comparing the two, but I thought perhaps there's a way to do it with a single query. Thanks again!

**Comment by antoinevulcain** [comment]
Yes for sure I just created an endpoint for that here is the link:

[https://financialmodelingprep.com/api/v3/stock/real-time-price-full/AAPL](https://financialmodelingprep.com/api/v3/stock/real-time-price-full/AAPL)

So you can have the data in a single query

**Comment by muziofg** [comment]
Very useful endpoint, especially the versione without ticker who listing all. Seems more reliable than gainers/losers endpoint to fetch that data.

I report you a bug:
changePercentage field is wrongly calculated over today price.
the formula used is: (100 * change) / price
but the correct formula must be: (100 * change) / (price - change)

Example who explain the bug (moreover, the symbol no longer exists):
https://financialmodelingprep.com/api/v3/stock/real-time-price-full/CNAC
last changePercentage is something like -244 

Also, this endpoint could be include the open, high, low and volume values for the last day. It could become an instant picture of the last "candle" 

Hoping to be useful

**Comment by antoinevulcain** [comment]
Here are a better endpoint for a real time change:

https://financialmodelingprep.com/api/v3/quote/AAPL,FB,GOOG,MSFT

**Comment by muziofg** [comment]
very good endpoint !
can you add "open" and "prevClose" fields to provide the open value of the day and the previous close value ?
Also a timestamp related to the current price could be useful to debugging purpose

**Comment by antoinevulcain** [comment]
@muziofg  Yes for the open and prevClose it should be possible, I look into it, could you explain more the timestamp that you need for the current price?

**Comment by muziofg** [comment]
an unequivocal timestamp, it can be a UNIX timestamp or a formatted date-time information like the standard "ISO 8601"

**Comment by antoinevulcain** [comment]
@muziofg  I added open, prevClose, timestamp in UNIX format, I also added EPS, PE , earningsAnnouncement and sharesOutstanding.

**Comment by muziofg** [comment]
ok, this endpoint is very close to the perfection
it's a very good work

A question: how many ticker can add for a single query?

It has two bugs:
- no data for the ticker SPCE (virgin galactic) https://financialmodelingprep.com/api/v3/quote/SPCE
- if no "earningsAnnouncement" exists yet for the future is correct to provide the last from the past? example: https://financialmodelingprep.com/api/v3/quote/SELB

thank you very much

**Comment by antoinevulcain** [comment]
@muziofg good point we will update the ticker list, SELB is a very recent stock that's why.

we decided to put the most recent earningsAnnouncement for developer to match the last 10-k / 10-q available on the API

For the number of ticker you can add for a sigle query their is no limitation yet

**Comment by antoinevulcain** [comment]
@muziofg so for the SELB you can now found it in the quote endpoint and historical endpoint:

https://financialmodelingprep.com/api/v3/quote/SELB

https://financialmodelingprep.com/api/v3/historical-price-full/SELB

We have now every 15 minutes a CRON running to check every new stock on every stock exchange and we automatically add it in the quote endpoint and add its historical data.

So now every ticker will be listed and if a new ticker (IPO) happens it will be added within 15 minutes to our database with the historical data and the real time update.

**Comment by muziofg** [comment]
good improvements!

but on change & changePercent/changePercentage fields I've some perplexity.

change value reported by https://financialmodelingprep.com/api/v3/quote/SELB is the difference between close and previous close values, that is what expected

change value reported by https://financialmodelingprep.com/api/v3/historical-price-full/SELB is the difference bewteen open and close values on same day (the size of a candle), that is not what expected to see on a field that has the same name (but different meaning) on other endpoints

**Comment by antoinevulcain** [comment]
@muziofg nice I see, you right.

Working on it now
